src/lib/falcor/FalcorController.js

```diff
@@ -34,41 +34,76 @@ export default class FalcorController extends BaseComponent {
   // this should always contain all the data the component needs
   loadFalcorCache() {
     console.log("LOADING FROM CACHE")
-    const falcorPath = this.constructor.getFalcorPath(this.props.params)
-
-    const data = this.props.model.getCache(falcorPath)
+    let falcorPath = this.constructor.getFalcorPath(this.props.params)
+    // If the component doesn't want any data
+    if (!falcorPath || falcorPath.length === 0) {
+      console.log("NO PATHS WERE GIVEN");
+      this.safeSetState({
+        ready: true,
+        data: []
+      })
+      return;
+    }
+    // If we're only passing a single pathSet we compensate for the spread operator
+    if (!(falcorPath[0] instanceof Array)) {
+      falcorPath = [falcorPath];
+    }
+    
+    const data = expandCache(this.props.model.getCache(...falcorPath));
     if (data) {
       this.safeSetState({
         ready: true,
         data: data
       })
     } else {
-      throw new Error("Serverside render of component: " + this.constructor.name +
-                      " failed. Data not in cache")
+      console.error("Serverside render of component: " + this.constructor.name +
```

Comment:
Build the error string once (with the falcor paths), and pass it into both places. I think you can even construct the error and do console.error(error)